easybuild/tools/job/slurm.py

```diff
@@ -0,0 +1,166 @@
+##
+# Copyright 2018-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Support for using Slurm as a backend for --job
+
+:author: Kenneth Hoste (Ghent University)
+"""
+import re
+from distutils.version import LooseVersion
+from vsc.utils import fancylogger
+
+from easybuild.tools.build_log import EasyBuildError, print_msg
+from easybuild.tools.config import JOB_DEPS_TYPE_ABORT_ON_ERROR, JOB_DEPS_TYPE_ALWAYS_RUN, build_option
+from easybuild.tools.job.backend import JobBackend
+from easybuild.tools.run import run_cmd
+
+
+_log = fancylogger.getLogger('slurm', fname=False)
+
+
+class Slurm(JobBackend):
+    """
+    Manage SLURM server communication and create `SlurmJob` objects.
+    """
+
+    REQ_VERSION = '17'
+
+    def __init__(self, *args, **kwargs):
+        """Constructor."""
+        super(Slurm, self).__init__(*args, **kwargs)
+
+        job_deps_type = build_option('job_deps_type')
+        if job_deps_type is None:
+            job_deps_type = JOB_DEPS_TYPE_ABORT_ON_ERROR
+            self.log.info("Using default job dependency type: %s", job_deps_type)
+        else:
+            self.log.info("Using specified job dependency type: %s", job_deps_type)
+
+        if job_deps_type == JOB_DEPS_TYPE_ABORT_ON_ERROR:
+            self.job_deps_type = 'afterok'
+        elif job_deps_type == JOB_DEPS_TYPE_ALWAYS_RUN:
+            self.job_deps_type = 'afterany'
+        else:
+            raise EasyBuildError("Unknown job dependency type specified: %s", job_deps_type)
+
+    def _check_version(self):
+        """Check whether version of Slurm complies with required version."""
+        (out, _) = run_cmd("sbatch --version", trace=False)
```

Comment:
This need a check on the return value too, in case sbatch doesn't exist in the PATH with a nice user visible error?

Reply:
`run_cmd` does that by default, if the exit code is non-zero it'll result is a clear error message

I can also check with `which('sbatch')` too though, to make sure the required commands are available...